.github/workflows/pr_checks.yml

```diff
@@ -93,7 +93,9 @@ jobs:
         uses: davelosert/vitest-coverage-report-action@v2
 
   ui-test:
-    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+    needs: [code_style, test]
+    if: ${{ always() && contains(join(needs.*.result, ','), 'success') && github.event.pull_request.draft == false }}
```

Comment:
`always() && contains(join(needs.*.result, ','), 'success')` what is this for? Do we need the `always()` ?